hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HealthTransformations.scala

```diff
@@ -0,0 +1,114 @@
+package org.broadinstitute.monster.dap
+
+import org.broadinstitute.monster.dogaging.jadeschema.table.HlesHealthCondition
+
+object HealthTransformations {
+
+  /** Parse all health-condition-related fields out of a raw RedCap record. */
+  def mapHealthConditions(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
+    mapInfectiousDisease(rawRecord)
+
+  /** Generic helper method for creating Hles Health Condition rows. */
+  def createHealthConditionRow(
+    rawRecord: RawRecord,
+    conditionName: String,
+    conditionType: Long,
+    condition: Long,
+    isCongenital: Boolean = false,
+    conditionOtherDescription: Option[String] = None,
+    conditionCause: Option[Long] = None,
+    conditionCauseOtherDescription: Option[String] = None
+  ): Option[HlesHealthCondition] =
+    if (rawRecord.getBoolean(s"hs_${conditionName}")) {
+      Some(
+        HlesHealthCondition(
+          dogId = rawRecord.getRequired("study_id").toLong,
+          hsConditionType = conditionType,
+          hsCondition = condition,
+          hsConditionOtherDescription = conditionOtherDescription,
+          hsConditionIsCongenital = isCongenital,
+          hsConditionCause = conditionCause,
+          hsConditionCauseOtherDescription = conditionCauseOtherDescription,
+          hsDiagnosisYear = rawRecord.getOptionalNumber(s"hs_${conditionName}_year"),
+          hsDiagnosisMonth = rawRecord.getOptionalNumber(s"hs_${conditionName}_month"),
+          hsRequiredSurgeryOrHospitalization =
+            rawRecord.getOptionalNumber(s"hs_${conditionName}_surg"),
+          hsFollowUpOngoing = rawRecord.getOptionalBoolean(s"hs_${conditionName}_fu")
+        )
+      )
+    } else {
+      None
+    }
+
+  /** Parse all infectious disease related fields out of a raw RedCap record. */
+  def mapInfectiousDisease(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
+    if (rawRecord.getBoolean("hs_dx_infectious_yn")) {
```

Comment:
I think this will need to use `getOptionalBoolean`

Reply:
If I'm reading the data dictionary correctly, this should always be present; it doesn't have a "Show the field ONLY if <condition>" listed

Reply:
My thinking for the goal is: We should be able to call the top-level `mapHealthConditions` method on `RawRecord(1, Map.empty)` and get out `HlesHealthCondition.init(1)`. As long as we can do that w/ this as `getBoolean` instead of `getOptionalBoolean`, I'm ok with it.

Reply:
I'll add that as a test